sympy/core/expr.py

```diff
@@ -1162,18 +1162,18 @@ def as_terms(self):
         for term in Add.make_args(self):
             coeff, _term = term.as_coeff_Mul()
 
-            coeff = complex(coeff)
+            coeff = [S.One, S.Zero]
+            def _coeff(c, d):
+                a, b = coeff
+                coeff[0] = a*c - d*b
+                coeff[1] = a*d + b*c
```

Comment:
Why is this arithmetic needed at all?

Reply:
To avoid evaluation via `complex`. 

The next question should be "Why are we trying to pay attention to the numerical portion (coefficient) of a term?" I think the answer is that SymPy was written with a bias towards interactive work and so the sorting which created something mathematically conventional was cooked into the core instead of being handled in the printing.

If we take out coefficient handling (beyond the leading `expr.as_coeff_Add()[0]`) we get
```python
>>> str((pi/10 + E*I).round(2))
'2.72*I + 0.31'
>>> ( 2 + 3*I).as_ordered_terms()
[3*I, 2]
```

Reply:
I can see that your changes here are an improvement but I just wonder if we can go any further with it.

I don't think that any actual arithmetic should be taking place here at all.

Splitting an Add with pure_complex seems okay but can we not then just distinguish an ordering without actual arithmetic?

Reply:
I am working on this.